iron-form.html

```diff
@@ -162,6 +171,21 @@
       this.fire('iron-form-submit', json);
     },
 
+    /**
+     * Resets all non-disabled elements to their initial values.
+     */
+    reset: function() {
+      // Trigger the native form validation, which in turn will also call
+      // `_resetCustomElements` to reset the form's custom element. Doing this
+      // prevents the event from being triggered twice.
+      HTMLFormElement.prototype.reset.call(this);
```

Comment:
You can manually call `ironForm.reset()`, in which case you need to reset both the custom and native elements. So in that case we just call the native reset, which will reset the things it knows about, and then trigger its `_onReset` listener, which will reset the custom elements.